swan-cmd/src/main/java/de/fraunhofer/iem/swan/features/code/cat/IsMethodFromLibraryFeature.java

```diff
@@ -0,0 +1,40 @@
+package de.fraunhofer.iem.swan.features.code.cat;
+
+import de.fraunhofer.iem.swan.data.Method;
+import de.fraunhofer.iem.swan.features.code.FeatureResult;
+import de.fraunhofer.iem.swan.features.code.ICodeFeature;
+
+import java.util.ArrayList;
+
+/***
+ * Evaluates whether the method is an application method.
+ *
+ * @author Rohith Kumar
+ */
+
+public class IsMethodFromLibraryFeature implements ICodeFeature {
+    private FeatureResult featureResult;
+    private ArrayList<String> featureValues;
+
+    public IsMethodFromLibraryFeature(){
+        this.featureResult = new FeatureResult();
+    }
+    @Override
+    public FeatureResult applies(Method method) {
+        this.featureResult.setBooleanValue(!method.isApplicationMethod());
```

Comment:
I think a better way to do this is to use the SootClass.isLibraryClass() instead of the check you use for application methods in the swan-cmd/src/main/java/de/fraunhofer/iem/swan/soot/Soot.java

Reply:
maybe we could even make this feature a categorical one being that we also have the SootClass.isJavaLibraryClass(). So the categories could be java library, library and application.

Reply:
Feature has been changed to a categorical feature.